### ISSUE TITLE
Error with fl+pox

### ISSUE THREAD

**Issue by KevinPhemius** [fp7-netide/Engine#10]
Hi,

I have this error when I launch floodlight over POX (with the UC3 application)

Here is floodlight log :

./floodlight-RB.sh
Starting floodlight server ...
OpenJDK Server VM warning: ignoring option PreBlockSpin=8; support was removed in 7.0_40
INFO [net.floodlightcontroller.core.module.FloodlightModuleLoader:main] Loading default modules
INFO [net.floodlightcontroller.core.internal.Controller:main] Controller role set to null
INFO [net.floodlightcontroller.interceptor.NetideModule:main] NetIDE Module binding to 41414...
INFO [net.floodlightcontroller.monitor.internal.Monitor:main] net.floodlightcontroller.monitor.internal.Monitor started
INFO [net.floodlightcontroller.core.internal.Controller:main] Listening for switch connections on 0.0.0.0/0.0.0.0:6634
INFO [net.floodlightcontroller.jython.JythonServer:debugserver-main] Starting DebugServer on port 6655
rawMessage size is 290
["packet", {"raw": [255, 255, 255, 255, 255, 255, 0, 0, 0, 0, 0, 3, 137, 66, 32, 0, 6, 4, 0, 2, 0, 0, 2, 7, 4, 0, 0, 0, 0, 0, 3, 4, 3, 2, 0, 3, 6, 2, 0, 120, 254, 12, 0, 38, 225, 0, 0, 0, 0, 0, 0, 0, 0, 3, 24, 8, 6, 19, 163, 131, 74, 7, 40, 245, 230, 1, 1, 0, 0], "switch": 5, "inport": 3}]
index and size of raw 9 279
java.lang.NullPointerException
        at net.floodlightcontroller.interceptor.BackendChannelHandler.sendMessageToController(BackendChannelHandler.java:206)
        at net.floodlightcontroller.interceptor.BackendChannelHandler.handleMessage(BackendChannelHandler.java:181)
        at net.floodlightcontroller.interceptor.BackendChannelHandler.messageReceived(BackendChannelHandler.java:73)
        at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:302)
        at org.jboss.netty.handler.codec.oneone.OneToOneDecoder.handleUpstream(OneToOneDecoder.java:76)
        at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:302)
        at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:317)
        at org.jboss.netty.handler.codec.frame.FrameDecoder.callDecode(FrameDecoder.java:299)
        at org.jboss.netty.handler.codec.frame.FrameDecoder.messageReceived(FrameDecoder.java:216)
        at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:274)
        at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:261)
        at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:351)
        at org.jboss.netty.channel.socket.nio.NioWorker.processSelectedKeys(NioWorker.java:282)
        at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:202)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)

And POX's :

python ./pox.py pox_client
POX 0.2.0 (carp) / Copyright 2011-2013 James McCauley, et al.
Connected to pyretic frontend.
INFO:core:POX 0.2.0 (carp) is up.
INFO:openflow.of_01:[None 1] closed
WARNING:openflow.of_01:<class 'pox.openflow.PacketIn'> raised on dummy OpenFlow nexus
WARNING:openflow.of_01:<class 'pox.openflow.PacketIn'> raised on dummy OpenFlow nexus
INFO:openflow.of_01:[00-00-00-00-00-05 2] connected
INFO:openflow.of_01:[00-00-00-00-00-06 3] connected
INFO:openflow.of_01:[00-00-00-00-00-07 4] connected
INFO:packet:(lldp tlv parse) warning unknown tlv type (12)
INFO:packet:(lldp tlv parse) warning unknown tlv type (115)

**Comment by KevinPhemius** [comment]
It looks like the LLDP packets are causing problems.

**Comment by antonis-m** [comment]
Let me reproduce it and I'll get back to you asap. I had tried to fix the issue with LLDP by using floodlight modules like the learning switch etc.. What I noticed (and we had this discussion with Roberto in Munich) was that floodlight encapsulates LLDP messages inside packetOut messages and therefore it was impossible to tell whether a PacketOut was a LLDP message or not without inspecting the payload of the packet. I didn't try it with the code for UC3 but I guess that it won't be that difficult to fix.
